### ISSUE TITLE
FTP - Failed to create file system for "target:/TEST": unsupported LIST line

### ISSUE THREAD

**Issue by alolis** [rclone/rclone#6879]
#### The associated forum post URL from `https://forum.rclone.org`

There is none.

#### What is the problem you are having with rclone?

I am getting `Failed to create file system for "target:/TEST": unsupported LIST line` error when I try to use the `lsd` command with my FTP backend and with any rclone version AFTER `v1.60.1`.

If I use `v1.60.1`, then everything works just fine.

#### What is your rclone version (output from `rclone version`)

```
rclone v1.62.2
- os/version: darwin 12.6 (64 bit)
- os/kernel: 21.6.0 (arm64)
- os/type: darwin
- os/arch: arm64 (ARMv8 compatible)
- go/version: go1.20.2
- go/linking: dynamic
- go/tags: cmount
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

MacOSX arm64

#### Which cloud storage system are you using? (e.g. Google Drive)

FTP

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

`rclone lsd target:/TEST`

The folder `TEST` exists on the remote.

And another note here, if I do `rclone lsd target:/TEST/NON_EXISTING_FOLDER` then I get a `Failed to create file system for "target:/TEST/NONE_EXISTING": 501 Not a valid pathname.` although it works just fine with `v1.60.1` and returns empty list instead.

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2023/03/22 19:32:23 DEBUG : rclone: Version "v1.62.2" starting with parameters ["rclone" "-vv" "--dump" "bodies" "--log-file" "bug.log" "--config" "rclone.conf" "lsd" "target:/TEST"]
2023/03/22 19:32:23 DEBUG : Creating backend with remote "target:/TEST"
2023/03/22 19:32:23 DEBUG : Using config file from "/Users/example/rclone.conf"
2023/03/22 19:32:23 DEBUG : ftp://example.com:21/TEST: Connecting to FTP server
2023/03/22 19:32:23 DEBUG : ftp://example.com:21/TEST: dial("tcp","example.com:21")
2023/03/22 19:32:23 DEBUG : ftp://example.com:21/TEST: > dial: conn=*fshttp.timeoutConn, err=<nil>
2023/03/22 19:32:23 DEBUG : FTP Rx: "220 Service ready for new user."
2023/03/22 19:32:23 DEBUG : FTP Tx: "USER example"
2023/03/22 19:32:23 DEBUG : FTP Rx: "331 User name okay, need password for example."
2023/03/22 19:32:23 DEBUG : FTP Tx: PASS *****
2023/03/22 19:32:25 DEBUG : FTP Rx: "230 User logged in, proceed."
2023/03/22 19:32:25 DEBUG : FTP Tx: "FEAT"
2023/03/22 19:32:25 DEBUG : FTP Rx: "211-Extensions supported"
2023/03/22 19:32:25 DEBUG : FTP Rx: " SIZE"
2023/03/22 19:32:25 DEBUG : FTP Rx: " MDTM"
2023/03/22 19:32:25 DEBUG : FTP Rx: " LANG en;zh-tw;ja;is"
2023/03/22 19:32:25 DEBUG : FTP Rx: " MLST Size;Modify;Create;Type;Perm"
2023/03/22 19:32:25 DEBUG : FTP Rx: " AUTH SSL"
2023/03/22 19:32:25 DEBUG : FTP Rx: " AUTH TLS"
2023/03/22 19:32:25 DEBUG : FTP Rx: " MODE Z"
2023/03/22 19:32:25 DEBUG : FTP Rx: " UTF8"
2023/03/22 19:32:25 DEBUG : FTP Rx: " TVFS"
2023/03/22 19:32:25 DEBUG : FTP Rx: " MD5"
2023/03/22 19:32:25 DEBUG : FTP Rx: " MMD5"
2023/03/22 19:32:25 DEBUG : FTP Rx: " MFCT"
2023/03/22 19:32:25 DEBUG : FTP Rx: " MFMT"
2023/03/22 19:32:25 DEBUG : FTP Rx: "211 End"
2023/03/22 19:32:25 DEBUG : FTP Tx: "TYPE I"
2023/03/22 19:32:25 DEBUG : FTP Rx: "200 Command TYPE okay."
2023/03/22 19:32:25 DEBUG : FTP Tx: "OPTS UTF8 ON"
2023/03/22 19:32:25 DEBUG : FTP Rx: "200 Command OPTS okay."
2023/03/22 19:32:25 DEBUG : FTP Tx: "MLST /TEST"
2023/03/22 19:32:26 DEBUG : FTP Rx: "250-"
2023/03/22 19:32:26 DEBUG : FTP Rx: "Size=2258880811416;Modify=20230322173224.000;Create=20221213132421.000;Type=dir; TEST"
2023/03/22 19:32:26 DEBUG : FTP Rx: ""
2023/03/22 19:32:26 DEBUG : FTP Rx: "250 Requested file action okay, completed."
2023/03/22 19:32:26 Failed to create file system for "target:/TEST": unsupported LIST line
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
> unsupported LIST line

That error means the ftp library had trouble parsing the listing line.

I put this one into the parser

> 2023/03/22 19:32:26 DEBUG : FTP Rx: "Size=2258880811416;Modify=20230322173224.000;Create=20221213132421.000;Type=dir; TEST"

And it worked fine.

I suspect that this blank line is causing the problem

> 2023/03/22 19:32:26 DEBUG : FTP Rx: ""

I think this is probably invalid output from the server.

Which server is it? It should say in the `-vv` log above but I think you've sanitized it.

It would be worth trying [the latest beta](https://beta.rclone.org/) to see if the recent ftp library update fixed it.

If not then this needs a fix to the upstream library.

Also - do you need to use leading `/`? Try your commands without a leading `/` and if they work use that.

**Comment by ncw** [comment]
Can you do a git bisection to find the exact commit it started to go wrong? That will give us a clue!

**Comment by alolis** [comment]
> Can you do a git bisection to find the exact commit it started to go wrong? That will give us a clue!

The latest exact working version is `v1.61.0-beta.6589.313493d51`. The problem appears in `v1.61.0-beta.6590.1628ca0d4` and more specifically after [this](https://github.com/rclone/rclone/commit/1628ca0d46b9ec313c425735ab85d11ec2695317) commit.

Based on the commit above, if I use the `----ftp-disable-mlsd` flag, everything works again (but I assume that performance will degrade).

**Comment by panthony** [comment]
@ncw I encountered an FTP server in the wild that was not compliant with the RFC 2389 and the maintainer of the upstream project was not particularly inclined to address the problem:

https://github.com/jlaffaye/ftp/issues/260

Which was unfortunate.

But your buggy response has not quite the same issue so maybe yours is more recent and/or more widespread with may change things.

**Comment by ncw** [comment]
I think this problem is caused by an extra blank line in the MLST output @panthony if you take a look at the first log in the thread

**Comment by danbeibei** [comment]
Same issue here with _Serveur FTP_ and _FTP Serveur WiFi_ Android apps as the FTP server

**Comment by danbeibei** [comment]
Sure: 
- [FTP Serveur WiFi](https://play.google.com/store/apps/details?id=com.medhaapps.wififtpserver&hl=fr&gl=US) 
- [Serveur FTP](https://play.google.com/store/apps/details?id=com.theolivetree.ftpserver&hl=fr&gl=US)

**Comment by ncw** [comment]
Thanks for the links @danbeibei very helpful.

I managed to reproduce this.

I've made a fix for the upstream library which I've temporarily forked.

Can you give this a go please @alolis and @danbeibei 

[v1.63.0-beta.6930.4bbebd7f1.fix-6879-ftp-list](https://beta.rclone.org/branch/fix-6879-ftp-list/v1.63.0-beta.6930.4bbebd7f1.fix-6879-ftp-list/) on branch [fix-6879-ftp-list](https://github.com/rclone/rclone/tree/fix-6879-ftp-list) (uploaded in 15-30 mins)

**Comment by danbeibei** [comment]
Thanks for the fix! It solves the `unsupported LIST line` error for me with the Android apps.

The second error @alolis described is still here though, but I don't know if it is related. It seems to fail creating new directories.
In this example the `tmp` directory does not exist on the remote `phone`. On the _fix-6879-ftp-list_ branch a `501 Not a valid pathname` error is returned:
```
$ rclone-v1.63.0-beta.6930.4bbebd7f1.fix-6879-ftp-list-linux-amd64/rclone copy -vv --dump bodies ./tmp phone:tmp
2023/03/28 07:37:17 DEBUG : rclone: Version "v1.63.0-beta.6930.4bbebd7f1.fix-6879-ftp-list" starting with parameters ["rclone-v1.63.0-beta.6930.4bbebd7f1.fix-6879-ftp-list-linux-amd64/rclone" "copy" "-vv" "--dump" "bodies" "./tmp" "phone:tmp"]
2023/03/28 07:37:17 DEBUG : Creating backend with remote "./tmp"
2023/03/28 07:37:17 DEBUG : Using config file from "/some/path/.config/rclone/rclone.conf"
2023/03/28 07:37:17 DEBUG : fs cache: renaming cache item "./tmp" to be canonical "/some/path/test/tmp"
2023/03/28 07:37:17 DEBUG : Creating backend with remote "phone:tmp"
2023/03/28 07:37:17 DEBUG : ftp://local_ip:port/tmp: Connecting to FTP server
2023/03/28 07:37:17 DEBUG : ftp://local_ip:port/tmp: dial("tcp","local_ip:port")
2023/03/28 07:37:17 DEBUG : ftp://local_ip:port/tmp: > dial: conn=*fshttp.timeoutConn, err=<nil>
2023/03/28 07:37:17 DEBUG : FTP Rx: "220 Service ready for new user."
2023/03/28 07:37:17 DEBUG : FTP Tx: "USER acces"
2023/03/28 07:37:17 DEBUG : FTP Rx: "331 User name okay, need password for acces."
2023/03/28 07:37:17 DEBUG : FTP Tx: PASS *****
2023/03/28 07:37:17 DEBUG : FTP Rx: "230 User logged in, proceed."
2023/03/28 07:37:17 DEBUG : FTP Tx: "FEAT"
2023/03/28 07:37:17 DEBUG : FTP Rx: "211-Extensions supported"
2023/03/28 07:37:17 DEBUG : FTP Rx: " SIZE"
2023/03/28 07:37:17 DEBUG : FTP Rx: " MDTM"
2023/03/28 07:37:17 DEBUG : FTP Rx: " REST STREAM"
2023/03/28 07:37:17 DEBUG : FTP Rx: " LANG en;zh-tw;ja;is"
2023/03/28 07:37:17 DEBUG : FTP Rx: " MLST Size;Modify;Type;Perm"
2023/03/28 07:37:17 DEBUG : FTP Rx: " AUTH SSL"
2023/03/28 07:37:17 DEBUG : FTP Rx: " AUTH TLS"
2023/03/28 07:37:17 DEBUG : FTP Rx: " MODE Z"
2023/03/28 07:37:17 DEBUG : FTP Rx: " UTF8"
2023/03/28 07:37:17 DEBUG : FTP Rx: " TVFS"
2023/03/28 07:37:17 DEBUG : FTP Rx: " MD5"
2023/03/28 07:37:17 DEBUG : FTP Rx: " MMD5"
2023/03/28 07:37:17 DEBUG : FTP Rx: " MFMT"
2023/03/28 07:37:17 DEBUG : FTP Rx: "211 End"
2023/03/28 07:37:17 DEBUG : FTP Tx: "TYPE I"
2023/03/28 07:37:17 DEBUG : FTP Rx: "200 Command TYPE okay."
2023/03/28 07:37:17 DEBUG : FTP Tx: "OPTS UTF8 ON"
2023/03/28 07:37:17 DEBUG : FTP Rx: "200 Command OPTS okay."
2023/03/28 07:37:17 DEBUG : FTP Tx: "MLST tmp"
2023/03/28 07:37:17 DEBUG : FTP Rx: "501 Not a valid pathname."
2023/03/28 07:37:17 DEBUG : FTP Tx: "NOOP"
2023/03/28 07:37:17 DEBUG : FTP Rx: "200 Command NOOP okay."
2023/03/28 07:37:17 Failed to create file system for "phone:tmp": 501 Not a valid pathname.
```
On version v1.60.1 the directory is successfully created and content copied:
```
$ rclone-v1.60.1-linux-amd64/rclone copy -vv --dump bodies ./tmp phone:tmp
2023/03/28 07:37:42 DEBUG : rclone: Version "v1.60.1" starting with parameters ["rclone-v1.60.1-linux-amd64/rclone" "copy" "-vv" "--dump" "bodies" "./tmp" "phone:tmp"]
2023/03/28 07:37:42 DEBUG : Creating backend with remote "./tmp"
2023/03/28 07:37:42 DEBUG : Using config file from "/some/path/.config/rclone/rclone.conf"
2023/03/28 07:37:42 DEBUG : fs cache: renaming cache item "./tmp" to be canonical "/some/path/test/tmp"
2023/03/28 07:37:42 DEBUG : Creating backend with remote "phone:tmp"
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: Connecting to FTP server
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: dial("tcp","local_ip:port")
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: > dial: conn=*fshttp.timeoutConn, err=<nil>
2023/03/28 07:37:42 DEBUG : FTP Rx: "220 Service ready for new user."
2023/03/28 07:37:42 DEBUG : FTP Tx: "USER acces"
2023/03/28 07:37:42 DEBUG : FTP Rx: "331 User name okay, need password for acces."
2023/03/28 07:37:42 DEBUG : FTP Tx: PASS *****
2023/03/28 07:37:42 DEBUG : FTP Rx: "230 User logged in, proceed."
2023/03/28 07:37:42 DEBUG : FTP Tx: "FEAT"
2023/03/28 07:37:42 DEBUG : FTP Rx: "211-Extensions supported"
2023/03/28 07:37:42 DEBUG : FTP Rx: " SIZE"
2023/03/28 07:37:42 DEBUG : FTP Rx: " MDTM"
2023/03/28 07:37:42 DEBUG : FTP Rx: " REST STREAM"
2023/03/28 07:37:42 DEBUG : FTP Rx: " LANG en;zh-tw;ja;is"
2023/03/28 07:37:42 DEBUG : FTP Rx: " MLST Size;Modify;Type;Perm"
2023/03/28 07:37:42 DEBUG : FTP Rx: " AUTH SSL"
2023/03/28 07:37:42 DEBUG : FTP Rx: " AUTH TLS"
2023/03/28 07:37:42 DEBUG : FTP Rx: " MODE Z"
2023/03/28 07:37:42 DEBUG : FTP Rx: " UTF8"
2023/03/28 07:37:42 DEBUG : FTP Rx: " TVFS"
2023/03/28 07:37:42 DEBUG : FTP Rx: " MD5"
2023/03/28 07:37:42 DEBUG : FTP Rx: " MMD5"
2023/03/28 07:37:42 DEBUG : FTP Rx: " MFMT"
2023/03/28 07:37:42 DEBUG : FTP Rx: "211 End"
2023/03/28 07:37:42 DEBUG : FTP Tx: "TYPE I"
2023/03/28 07:37:42 DEBUG : FTP Rx: "200 Command TYPE okay."
2023/03/28 07:37:42 DEBUG : FTP Tx: "OPTS UTF8 ON"
2023/03/28 07:37:42 DEBUG : FTP Rx: "200 Command OPTS okay."
2023/03/28 07:37:42 DEBUG : FTP Tx: "EPSV"
2023/03/28 07:37:42 DEBUG : FTP Rx: "229 Entering Passive Mode (|||41829|)"
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: dial("tcp","local_ip:41829")
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: > dial: conn=*fshttp.timeoutConn, err=<nil>
2023/03/28 07:37:42 DEBUG : FTP Tx: "MLSD ."
2023/03/28 07:37:42 DEBUG : FTP Rx: "150 File status okay; about to open data connection."
2023/03/28 07:37:42 DEBUG : FTP Rx: "Size=3452;Modify=20220523120420.000;Type=dir; some_dir_0"
2023/03/28 07:37:42 DEBUG : FTP Rx: "Size=3452;Modify=20221202212628.000;Type=dir; some_dir_1"
2023/03/28 07:37:42 DEBUG : FTP Rx: "Size=3452;Modify=20230225111451.000;Type=dir; some_dir_3"
2023/03/28 07:37:42 DEBUG : FTP Rx: "226 Closing data connection."
2023/03/28 07:37:42 DEBUG : FTP Tx: "EPSV"
2023/03/28 07:37:42 DEBUG : FTP Rx: "229 Entering Passive Mode (|||40243|)"
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: dial("tcp","local_ip:40243")
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: > dial: conn=*fshttp.timeoutConn, err=<nil>
2023/03/28 07:37:42 DEBUG : FTP Tx: "MLSD tmp"
2023/03/28 07:37:42 DEBUG : FTP Rx: "150 File status okay; about to open data connection."
2023/03/28 07:37:42 DEBUG : FTP Rx: "226 Closing data connection."
2023/03/28 07:37:42 DEBUG : FTP Tx: "EPSV"
2023/03/28 07:37:42 DEBUG : FTP Rx: "229 Entering Passive Mode (|||43263|)"
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: dial("tcp","local_ip:43263")
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: > dial: conn=*fshttp.timeoutConn, err=<nil>
2023/03/28 07:37:42 DEBUG : FTP Tx: "MLSD ."
2023/03/28 07:37:42 DEBUG : FTP Rx: "150 File status okay; about to open data connection."
2023/03/28 07:37:42 DEBUG : FTP Rx: "Size=3452;Modify=20220523120420.000;Type=dir; some_dir_0"
2023/03/28 07:37:42 DEBUG : FTP Rx: "Size=3452;Modify=20221202212628.000;Type=dir; some_dir_1"
2023/03/28 07:37:42 DEBUG : FTP Rx: "Size=3452;Modify=20230225111451.000;Type=dir; some_dir_3"
2023/03/28 07:37:42 DEBUG : FTP Rx: "226 Closing data connection."
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: Waiting for checks to finish
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: Waiting for transfers to finish
2023/03/28 07:37:42 DEBUG : FTP Tx: "EPSV"
2023/03/28 07:37:42 DEBUG : FTP Rx: "229 Entering Passive Mode (|||38013|)"
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: dial("tcp","local_ip:38013")
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: > dial: conn=*fshttp.timeoutConn, err=<nil>
2023/03/28 07:37:42 DEBUG : FTP Tx: "MLSD ."
2023/03/28 07:37:42 DEBUG : FTP Rx: "150 File status okay; about to open data connection."
2023/03/28 07:37:42 DEBUG : FTP Rx: "Size=3452;Modify=20220523120420.000;Type=dir; some_dir_0"
2023/03/28 07:37:42 DEBUG : FTP Rx: "Size=3452;Modify=20221202212628.000;Type=dir; some_dir_1"
2023/03/28 07:37:42 DEBUG : FTP Rx: "Size=3452;Modify=20230225111451.000;Type=dir; some_dir_3"
2023/03/28 07:37:42 DEBUG : FTP Rx: "226 Closing data connection."
2023/03/28 07:37:42 DEBUG : FTP Tx: "MKD tmp"
2023/03/28 07:37:42 DEBUG : FTP Rx: "257 \"/tmp\" created."
2023/03/28 07:37:42 DEBUG : FTP Tx: "EPSV"
2023/03/28 07:37:42 DEBUG : FTP Rx: "229 Entering Passive Mode (|||38507|)"
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: dial("tcp","local_ip:38507")
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: > dial: conn=*fshttp.timeoutConn, err=<nil>
2023/03/28 07:37:42 DEBUG : FTP Tx: "STOR tmp/test.file"
2023/03/28 07:37:42 DEBUG : FTP Rx: "150 File status okay; about to open data connection."
2023/03/28 07:37:42 DEBUG : FTP Rx: "226 Transfer complete."
2023/03/28 07:37:42 DEBUG : FTP Tx: "MFMT 20230328051540 tmp/test.file"
2023/03/28 07:37:42 DEBUG : FTP Rx: "213 ModifyTime=20230328051540; tmp/test.file"
2023/03/28 07:37:42 DEBUG : FTP Tx: "EPSV"
2023/03/28 07:37:42 DEBUG : FTP Rx: "229 Entering Passive Mode (|||38201|)"
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: dial("tcp","local_ip:38201")
2023/03/28 07:37:42 DEBUG : ftp://local_ip:port/tmp: > dial: conn=*fshttp.timeoutConn, err=<nil>
2023/03/28 07:37:42 DEBUG : FTP Tx: "MLSD tmp"
2023/03/28 07:37:42 DEBUG : FTP Rx: "150 File status okay; about to open data connection."
2023/03/28 07:37:43 DEBUG : FTP Rx: "Size=0;Modify=20230328051540.000;Type=file; test.file"
2023/03/28 07:37:43 DEBUG : FTP Rx: "226 Closing data connection."
2023/03/28 07:37:43 INFO  : test.file: Copied (new)
2023/03/28 07:37:43 INFO  :
Transferred:              0 B / 0 B, -, 0 B/s, ETA -
Transferred:            1 / 1, 100%
Elapsed time:         0.7s

2023/03/28 07:37:43 DEBUG : 3 go routines active
2023/03/28 07:37:43 DEBUG : ftp://local_ip:port/tmp: closing 1 unused connections
2023/03/28 07:37:43 DEBUG : FTP Tx: "QUIT"
```

**Comment by ncw** [comment]
The server appears to be returning the wrong error code for `MLST doesnotexist`. However the RFC isn't outstandingly clear here and does mention the 501 error code in relation to the related MLSD command so it probably isn't terribly wrong!

> Giving a pathname that exists but is not a directory as the argument to a MLSD command generates a 501 reply.

It is easy enough to mask that error though, can you try this?

[v1.63.0-beta.6931.9639c447e.fix-6879-ftp-list](https://beta.rclone.org/branch/fix-6879-ftp-list/v1.63.0-beta.6931.9639c447e.fix-6879-ftp-list/) on branch [fix-6879-ftp-list](https://github.com/rclone/rclone/tree/fix-6879-ftp-list) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I have reported a bug https://github.com/jlaffaye/ftp/pull/321 and made a PR to fix the bug https://github.com/jlaffaye/ftp/pull/322 in the upstream library.

**Comment by alolis** [comment]
@ncw , I confirm that the version from [https://beta.rclone.org/branch/fix-6879-ftp-list/v1.63.0-beta.6931.9639c447e.fix-6879-ftp-list/](https://beta.rclone.org/branch/fix-6879-ftp-list/v1.63.0-beta.6931.9639c447e.fix-6879-ftp-list/) worked.

As for the other issue, the `501 Not a valid pathname` when I issue an `lsd` on a non-existing path I now see a slightly different behavior. Although the error does appear in the log (which I find normal), the operation completes without the `Failed to create file system for ....` error I was getting before so I assume everything is in order now.

Excellent work I must say!

**Comment by danbeibei** [comment]
Works for me to, thanks a lot

**Comment by panthony** [comment]
@ncw Thanks for this, sorry I'm a little overwhelmed at the moment

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.63

I've forked the upstream again since they haven't merged my patch.

**Comment by alolis** [comment]
@ncw, thank you for your efforts.
